open-sphere-plugins/infinity/src/main/java/io/opensphere/infinity/envoy/InfinityEnvoy.java

```diff
@@ -0,0 +1,286 @@
+    {
+        URL url = getUrl(category);
+        LOGGER.info("Request: " + url + " " + parameters.getTimeSpan() + " " + parameters.getGeometry());
+
+        InputStream postData = createRequestStream(parameters);
+        ResponseValues response = new ResponseValues();
+        ServerProvider<HttpServer> provider = getServerProviderRegistry().getProvider(HttpServer.class);
+
+        try (CancellableInputStream inputStream = HttpUtilities.sendPost(url, postData, response, ContentType.JSON, provider))
+        {
+            Collection<SearchResponse> items = parseDepositItems(inputStream);
+            if (!items.isEmpty())
+            {
+                CacheDeposit<SearchResponse> deposit = createDeposit(category, items, parameters);
+                queryReceiver.receive(deposit);
+            }
+        }
+    }
+
+    /**
+     * Creates a JSON request stream (the post body).
+     *
+     * @param parameters the query parameters
+     * @return the request stream
+     * @throws IOException if something goes wrong
+     */
+    private InputStream createRequestStream(QueryParameters parameters) throws IOException
+    {
+        SearchRequest request = createSearchRequest(parameters);
+
+        ByteArrayOutputStream out = new ByteArrayOutputStream(384);
+        ObjectMapper mapper = JsonUtils.createMapper();
+        mapper.setSerializationInclusion(JsonSerialize.Inclusion.NON_NULL);
+        mapper.writeValue(out, request);
+
+        if (LOGGER.isDebugEnabled())
+        {
+            LOGGER.debug(new String(out.toByteArray()));
+        }
+
+        InputStream postData = new ByteArrayInputStream(out.toByteArray());
+        return postData;
+    }
+
+    /**
+     * Creates the search request bean.
+     *
+     * @param parameters the query parameters
+     * @return the search request bean
+     */
+    private SearchRequest createSearchRequest(QueryParameters parameters)
+    {
+        SearchRequest request = new SearchRequest();
+        request.setSize(0);
+        request.setTimeout("30s");
+        Object[] must = new Object[2];
+        if (parameters.getEndTimeField() != null)
+        {
+            Bool bool = new Bool();
+            Object[] innerMust = new Object[2];
+            innerMust[0] = new Any("range",
+                    new Any(parameters.getTimeField(), new TimeRange(null, Long.valueOf(parameters.getTimeSpan().getEnd()))));
+            innerMust[1] = new Any("range", new Any(parameters.getEndTimeField(),
+                    new TimeRange(Long.valueOf(parameters.getTimeSpan().getStart()), null)));
+            bool.setMust(innerMust);
+            must[0] = new Any("bool", bool);
+        }
+        else
+        {
+            must[0] = new Any("range", new Any(parameters.getTimeField(), new TimeRange(parameters.getTimeSpan())));
+        }
+        if (parameters.getGeometryType() == GeometryType.POINT)
+        {
+            must[1] = new Any("geo_bounding_box",

```

Comment:
Picking a random line but maybe pull these strings into constants.

Reply:
I actually prefer to have these here - it seems easier to update if the API ever changes